### PR TITLE
Fixes for issue #90

### PR DESCRIPTION
**Pull request by maeriens**
Here I fix what I mentioned in #90, and also rebased from #89 so everything can go in one single PR.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -168,38 +170,41 @@ class StickyTable extends React.Component {
...
+        }
       }
 
       stickyInsets[stickyKey] = insets;
     });
 
     //HINT avoid a render unless there's actually a change
     if (JSON.stringify(stickyInsets) !== JSON.stringify(this.state.stickyInsets)) {
-      this.setState({stickyInsets});
+      this.setState({ stickyInsets });
```
Comment (maxhudson) [inline src/index.js:198]: lots of opinionated styles mixed in with a functional pull request - I'm personally not a fan of the spacing style changes made in this PR, but do appreciate the functional code

File: src/index.js [hunk 2]
```diff
@@ -76,6 +76,8 @@ const Wrapper = styled('div').attrs({
     var insets = props.stickyInsets;
     var styles = '';
     var i;
+    console.log(props)
```
Comment (maxhudson) [inline src/index.js:79]: remove console logs

File: src/index.js [hunk 3]
```diff
@@ -24,17 +24,17 @@ const Cell = styled('div').attrs({
...
   className: 'sticky-table-row'
-})`
+}))`
   display: table-row;
 `;
 
 Row.displayName = 'Row';
 
-const Wrapper = styled('div').attrs({
+const Wrapper = styled('div').attrs(() => ({
```
Comment (maxhudson) [inline src/index.js:35]: what's the reason for this change?

File: src/index.js [hunk 4]
```diff
@@ -168,38 +170,41 @@ class StickyTable extends React.Component {
       ['leftColumn', 'width', 'leftStickyColumnCount'],
       ['rightColumn', 'width', 'rightStickyColumnCount']
     ].forEach(([stickyKey, sizeKey, countPropKey]) => {
-      var insets = [0];
-      var count = props[countPropKey];
-      var netInset = 0;
+      var insets = [];
```
Comment (maxhudson) [inline src/index.js:173]: seems odd to have `insets[0]` be `undefined` for count 1, and `insets[0]` be 0 for count 2+

File: src/index.js [hunk 5]
```diff
@@ -168,38 +170,41 @@ class StickyTable extends React.Component {
...
-          var boundingRect = node.getBoundingClientRect();
+        // HINT we only want this loop for the second sticky and up
+        for (s = 1; s < count; s++) {
+          var node = stickyKey === 'header' || stickyKey === 'leftColumn' ? cellNodes[s - 1] : cellNodes[cellNodes.length - s];
 
-          netInset += boundingRect[sizeKey];
-        }
+          if (node) {
+            var boundingRect = node.getBoundingClientRect();
+            netInset += boundingRect[sizeKey];
```
Comment (maxhudson) [inline src/index.js:186]: new line above 186

File: src/index.js (hunk 4 above)
Comment (maxhudson) [reply src/index.js:173]: implying this should be `var insets = [0]`

File: src/index.js [hunk 6]
```diff
@@ -168,38 +170,41 @@ class StickyTable extends React.Component {
...
+      if (props[countPropKey] > 1) {
+        insets = [0]
+        var count = props[countPropKey];
+        var netInset = 0;
 
-        if (node) {
-          var boundingRect = node.getBoundingClientRect();
+        // HINT we only want this loop for the second sticky and up
+        for (s = 1; s < count; s++) {
+          var node = stickyKey === 'header' || stickyKey === 'leftColumn' ? cellNodes[s - 1] : cellNodes[cellNodes.length - s];
```
Comment (maxhudson) [inline src/index.js:182]: this seems like the one significant change here - definitely should be doing something with s, and not sure how I missed that, but I'm pretty sure this isn't quite right. need to be doing something with rows and then cells both based on s (this is only accounting for more than 2 columns, but not more than 2 footers/headers)

File: src/index.js (hunk 1 above)
Comment (maeriens) [reply src/index.js:198]: I can revert them if you want - I think I have a global styling that somehow overwrote the `.eslint` or I used the format on save from vscode; not sure

File: src/index.js (hunk 3 above)
Comment (maeriens) [reply src/index.js:35]: It was deprecated on `styled-components@4` but was backwards compatible - and dropped in v5 - it's point 7 in their FAQ here https://styled-components.com/docs/faqs#what-do-i-need-to-do-to-migrate-to-v4

**Comment by maeriens** [comment]
@maxhudson do you want me to close this one and send again the change with only the storybook addons, and then in here continue the other changes?

File: src/index.js (hunk 4 above)
Comment (maeriens) [reply src/index.js:173]: The first solution I had was to declare `var insets = props[countPropKey] > 1 ? [0] : [];` because the issue is that it sets to 0 all sides when any of the other stickyes is higher than 1

File: src/index.js (hunk 6 above)
Comment (maeriens) [reply src/index.js:182]: Hmm. You're right. This fixes columns but not the others.
Maybe getting the `rowNodes` from the table too? That way, if the `stickyKey` is `header` or `footer`, it should check against the cells (probably `rowNodes[s - 1] : rowNodes[rowNodes.length - s];?)

**Comment by maxhudson** [comment]
@maeriens I think it's fine to leave it in one place for now. Will give fixing the row/cell node indices a go tomorrow

**Comment by maeriens** [comment]
Removed the spaces where they were unwanted - and added a patch that worked for all tests I ran (one sticky in one, two or three places, up to 4 stickies in one, two or three directions).
We can improve over this now - I didn't have time to give it a bit more of thought to come up with a stronger solution today

File: src/index.js [hunk 7]
```diff
@@ -168,21 +169,40 @@ class StickyTable extends React.Component {
...
+        // HINT we only want this loop for the second sticky and up
+        for (s = 1; s < count; s++) {
+          var node = undefined;
+
+          switch (stickyKey) {
+            case 'header':
+              node = rowNodes[s - 1]
+              break;
+            case 'footer':
+              node = rowNodes[rowNodes.length - s]
```
Comment (maxhudson) [inline src/index.js:188]: when the node is a row, we should be using childNodes[0] so we're always getting size of a cell - in case there's a browser bug with table row height

File: src/index.js (hunk 4 above)
Comment (maxhudson) [reply src/index.js:172]: I think my first comment in this thread is still relevant or the answer is unclear to me. I still don't understand why it should ever be empty. 

I actually wrote the code so that 0 would never be necessary here, so guessing insets should never have 0 in it, unless a cell has width 0. The first cell on a given side will always have an inset of 0 regardless of the status of the other sticky rows/columns

Comment (maxhudson) [reply src/index.js:172]: seems like 0 should always be in insets, or never be in it

**Comment by maxhudson** [comment]
I'm going to merge this as it's close enough and I'll play around with it/clean up on my end before publishing.

**Comment by maxhudson** [comment]
Thank you very much for the help @maeriens  - let me know if I can buy you a coffee!

**Comment by maxhudson** [comment]
@maeriens I couldn't get your playground to work - not sure if it's something on my end or just a misconfiguration. Assuming expected behavior is resizable column handles?

**Comment by maeriens** [comment]
@maxhudson sorry I didn't answer before - no problemo over the coffee, I think this is my first real contribution to something useful. As for the playgorund, I just run `yarn run storybook` and it displays correctly the knobs in that story only (not in the rest). Here is a gif of it.
image

**Comment by maxhudson** [comment]
Weird - not sure how to get that to show up. Doesn't seem to be working here: https://henrybuilt.github.io/react-sticky-table/?path=/story/basic--playground

**Comment by maxhudson** [comment]
@maeriens out of curiosity - are you using the package for something? Would be great to have some example links!

**Comment by maeriens** [comment]
Ah - I think the tab is not open by default - press `D` in the keyboard to bring it up. It an be configured, I forgot

**Comment by maxhudson** [comment]
ah cool - thanks
